p2panda-rs/src/secret_group/commit.rs

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use openmls::framing::{
+    MlsMessageIn, MlsMessageOut, MlsPlaintextContentType, VerifiableMlsPlaintext,
+};
+use openmls::messages::Welcome;
+use tls_codec::{Size, TlsDeserialize, TlsSerialize, TlsSize};
+
+use crate::secret_group::{SecretGroupError, SecretGroupMessage};
+
+/// Plaintext commit message which is published on the network to announce group changes.
+///
+/// A `SecretGroupCommit` always contains a MLS Commit message and the current, encrypted long term
+/// secrets for this group. Optionally it contains another MLS Welcome message in case this commit
+/// invites new members into the group.
+#[derive(Debug, Clone, TlsSerialize, TlsDeserialize, TlsSize)]
+pub struct SecretGroupCommit {
+    mls_commit_message: VerifiableMlsPlaintext,
+    mls_welcome_message: Option<Welcome>,
+    encrypted_long_term_secrets: SecretGroupMessage,
+}
+
+impl SecretGroupCommit {
+    /// Returns a new instance of a [SecretGroupCommit] message.
+    pub(crate) fn new(
+        mls_message_out: MlsMessageOut,
+        mls_welcome_message: Option<Welcome>,
+        encrypted_long_term_secrets: SecretGroupMessage,
+    ) -> Result<Self, SecretGroupError> {
+        // Check if message is in plaintext
+        let mls_commit_message = match mls_message_out.clone() {
+            MlsMessageOut::Plaintext(message) => Ok(message),
+            _ => Err(SecretGroupError::NeedsToBeMlsPlaintext),
```

Comment:
You shouldn't have to to do this check, I filed https://github.com/openmls/openmls/issues/584 for that

Reply:
Ok, super! I'll keep an issue about it for now :+1: I can imagine this could be set in the `ManagedGroupConfig`, maybe even aligned with the `.wire_format` setting?